### PR TITLE
Added SHIPPING_OWN to address list command

### PR DESCRIPTION
**Pull request by lazaruscrypto**
PD-158. Added SHIPPING_OWN and gui title check to address list command and created tests.

### CODE REVIEW

File: src/api/commands/address/AddressListCommand.ts [hunk 1]
```diff
@@ -38,13 +40,18 @@ export class AddressListCommand extends BaseCommand implements RpcCommandInterfa
...
 
-        // Get only addresses associated with a given profile ID.
         const profile = await this.profileService.findOne(profileId, true);
+        const type = data.params[1] ? data.params[1] : AddressType.SHIPPING_OWN;
+
+        // Return SHIPPING_OWN addresses by default
+        if (type === AddressType.SHIPPING_OWN)  {
+            return profile.toJSON().ShippingAddresses.filter((address) => address.title && address.type === 'SHIPPING_OWN');
+        }
         return profile.toJSON().ShippingAddresses;
```
[the remark this answers is not in the document]
Comment (xludx) [reply src/api/commands/address/AddressListCommand.ts]: and you return addresses that have no title, I dont think our api needs to have an opinion on whether addresses need title or not, GUI can decide whether they want that and validate the data before sending. if we wouldnt want addresses with title, we should rather make the field not null so it wouldnt be possible to save them and would make filtering them out useless.

File: src/api/commands/address/AddressListCommand.ts [hunk 2]
```diff
@@ -38,13 +40,18 @@ export class AddressListCommand extends BaseCommand implements RpcCommandInterfa
     @validate()
     public async execute( @request(RpcRequest) data: RpcRequest, rpcCommandFactory: RpcCommandFactory): Promise<Bookshelf.Collection<Address>> {
         const profileId = data.params[0];
-        if ( !profileId ) {
+        if (!profileId) {
             // Get all addresses.
             return this.addressService.findAll();
         }
```
Comment (xludx) [inline src/api/commands/address/AddressListCommand.ts]: profileId is not optional, if there's no profile id, throw error.

Comment (vlazlatoken) [reply src/api/commands/address/AddressListCommand.ts]: Then this test must be fixed - https://github.com/particl/particl-market/blob/develop/test/black-box/address/AddressRemoveCommand.test.ts#L64

Comment (lazaruscrypto) [reply src/api/commands/address/AddressListCommand.ts]: Then we need to fix this test https://github.com/particl/particl-market/blob/develop/test/black-box/address/AddressRemoveCommand.test.ts#L64

Comment (xludx) [reply src/api/commands/address/AddressListCommand.ts]: yes, remove that and you can add test to make sure address list returns the addresses for the correct profile

Comment (lazaruscrypto) [reply src/api/commands/address/AddressListCommand.ts]: I'll simply add addressId as arg there, so that it will return empty res.

**Pull request merged**